packages/swing-store/src/exporter.js

```diff
@@ -170,14 +164,22 @@ export function makeSwingStoreExporter(dirPath, options = {}) {
 
   /**
    * @returns {AsyncIterableIterator<string>}
-   * @yields {string}
    */
-  async function* getArtifactNames() {
-    yield* snapStore.getArtifactNames(artifactMode);
-    yield* transcriptStore.getArtifactNames(artifactMode);
-    yield* bundleStore.getArtifactNames();
+  function getArtifactNames() {
+    if (artifactMode !== 'debug') {
+      // throw if this DB will not be able to yield all the desired artifacts
+      const internal = { snapStore, bundleStore, transcriptStore };
+      assertComplete(internal, artifactMode);
+    }
+    // weird construct here is because we need to be able to throw before the generator starts
+    async function* exporter() {
+      yield* snapStore.getArtifactNames(artifactMode);
+      yield* transcriptStore.getArtifactNames(artifactMode);
+      yield* bundleStore.getArtifactNames();
+    }
+    harden(exporter);
```

Comment:
To match the protective behavior of the original version, we still need to `harden(getArtifactNames)`, otherwise two isolated importers of the module could use it to communicate.

I don't think `harden(exporter)` does anything useful: we don't return it, and the caller has no way to access it.

I don't know whether it would be useful to harden the *iterator* that is returned by this function.. the original did not do that, and I think it doesn't represent a communication channel (since each invocation of `getArtifactNames()` returns a different iterator).

Reply:
That is not actually correct.
- `exporter` needs to be hardened because it's actually available through the iterator object. Generators are like defining a class.
- `getArtifactNames` as a generator was previously explicitly hardened for that reason. We don't have a lint rule about explicitly hardening generators, but we probably should, if we could. However it was always redundant since it was getting hardened as a method of the `Exporter` object returned by this maker. Because it's now a simple non-generator function, we can rely on that harden.

Reply:
I actually didn't need it to be in the closure, so extracted. It should be slightly less clunky

Reply:
ah, ok, I didn't realize they were reachable.. sounds good